src/modules/usart.rs

```diff
@@ -0,0 +1,258 @@
+use core::prelude::*;
+use core::default::Default;
+
+use collections::vec::Vec;
+use collections::string::String;
+
+use {cmu, gpio, usart};
+use gpio::Port;
+
+use self::Location::*;
+
+type PortPin = Option<(Port, u32)>;
+
+/// Possible locations for the different Usart's. Not all locations is available for every Usart.
+#[derive(Copy, Debug)]
+pub enum Location {
+    Loc0,
+    Loc1,
+    Loc2,
+    Loc3,
+    Loc4,
+    Loc5,
+}
+
+/// Possible different Usart's to configure.
+#[derive(Copy, Debug)]
+pub enum Config {
+    Usart0(Location),
+    Usart1(Location),
+    Usart2(Location),
+}
+
+#[allow(dead_code)]
+pub struct Usart {
+    config: Config,
+    baudrate: u32,
+    usart: &'static mut usart::Usart,
+    line_break: &'static str,
+}
+
+/// Initializes and returns a new Usart1 instance with Location 1 and a baudrate of 9600.
+impl Default for Usart {
+    fn default() -> Usart {
+        let mut usart = Usart {
+            config: Config::Usart1(Loc1),
+            baudrate: 9600,
+            usart: usart::Usart::usart1(),
+            line_break: "\0\r",
+        };
+
+        usart.init();
```

Comment:
I suggest using an extra constructor function for this and not the default trait.